src/components/MultiGestureCanvas/usePanGesture.ts

```diff
@@ -119,88 +119,96 @@ const usePanGesture = ({
+                            runOnJS(onSwipeDown)();
+                        }
+                    }),
+                );
             } else {
                 // Animated back to the boundary
-                offsetY.value = withSpring(clampedOffset.y, SPRING_CONFIG, () => {
-                    isSwipingDownToClose.value = false;
-                });
+                offsetY.set(
+                    withSpring(clampedOffset.y, SPRING_CONFIG, () => {
+                        isSwipingDownToClose.set(false);
+                    }),
+                );
             }
         }
 
         // Reset velocity variables after we finished the pan gesture
-        panVelocityX.value = 0;
-        panVelocityY.value = 0;
-    }, [offsetX, offsetY, panTranslateX, panTranslateY, panVelocityX, panVelocityY, zoomScale, isSwipingDownToClose, getBounds, onSwipeDown]);
+        panVelocityX.set(0);
+        panVelocityY.set(0);
+    }, [getBounds, isSwipingDownToClose, offsetX, offsetY, onSwipeDown, panTranslateX, panTranslateY, panVelocityX, panVelocityY, zoomScale]);
 
     const panGesture = Gesture.Pan()
         .manualActivation(true)
         .averageTouches(true)
         .onTouchesUp(() => {
-            previousTouch.value = null;
+            previousTouch.set(null);
         })
         .onTouchesMove((evt, state) => {
             // We only allow panning when the content is zoomed in
-            if (zoomScale.value > 1 && !shouldDisableTransformationGestures.value) {
+            if (zoomScale.get() > 1 && !shouldDisableTransformationGestures.get()) {
                 state.activate();
             }
 
             // TODO: this needs tuning to work properly
-            if (!shouldDisableTransformationGestures.value && zoomScale.value === 1 && previousTouch.value !== null) {
-                const velocityX = Math.abs((evt.allTouches.at(0)?.x ?? 0) - previousTouch.value.x);
-                const velocityY = (evt.allTouches.at(0)?.y ?? 0) - previousTouch.value.y;
+            const previousTouchValue = previousTouch.get();
+            if (!shouldDisableTransformationGestures.get() && zoomScale.get() === 1 && previousTouchValue !== null) {
+                const velocityX = Math.abs((evt.allTouches.at(0)?.x ?? 0) - previousTouchValue.x);
+                const velocityY = (evt.allTouches.at(0)?.y ?? 0) - previousTouchValue.y;
 
                 if (Math.abs(velocityY) > velocityX && velocityY > 20) {
                     state.activate();
 
-                    // eslint-disable-next-line react-compiler/react-compiler
-                    isSwipingDownToClose.value = true;
-                    previousTouch.value = null;
+                    isSwipingDownToClose.set(true);
+                    previousTouch.set(null);
 
                     return;
                 }
             }
 
-            if (previousTouch.value === null) {
-                previousTouch.value = {
+            if (previousTouchValue === null) {
```

Comment:
`previousTouchValue` might be outdated
```suggestion
            if (previousTouch.get() === null) {
```